mirrord/kube/src/resolved/cron_job.rs

```diff
@@ -0,0 +1,48 @@
+use std::collections::BTreeMap;
+
+use k8s_openapi::api::batch::v1::CronJob;
+
+use super::ResolvedResource;
+use crate::{
+    api::runtime::RuntimeDataFromLabels,
+    error::{KubeApiError, Result},
+};
+
+#[derive(Debug, Clone)]
+pub struct ResolvedCronJob {
+    pub resource: CronJob,
+    pub container: Option<String>,
+}
+
+impl RuntimeDataFromLabels for ResolvedResource<CronJob> {
+    type Resource = CronJob;
+
+    fn name(&self) -> &str {
+        self.resource.metadata.name.as_ref().unwrap().as_str()
```

Comment:
This method can be implemented in the trait definition instead of in all implementations, as the associated type has a bound to `Resource`.

Reply:
Only if we have a method `resource` in the trait, so we can do `self.resource().meta().*`, but the `config::Target` stuff doesn't hold the `Resource`, so it can't really be done.

e.g.

```rust
impl RuntimeDataFromLabels for RolloutTarget {
    type Resource = Rollout;

    fn resource(&self) -> &Self::Resource { 
        // what could go here?
        // self.resource <- only exists for the `Resource<Rollout>` type
    }
```